docs/source/audit/implementation.rst

```diff
@@ -0,0 +1,149 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+*audit.query* performs validation when the API user has specified `select`
+query-options to ensure that the parameter being selected is actually provided
+somewhere in our audit entries. This is to help users avoid generating malformed
+audit information reports. The set of valid parameters must be expanded to
+include the parameters from the new audited service.
+
+7. Update the service to generate audit messages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+7.1. Sending syslog message
+The syslog-ng service filters audit messages based on the ident of the message
+received so that they are inserted into the correct audit database. The service
+in this case must set the ident to TNAUDIT_{service name}, for example `TNAUDIT_SMB`.
+
+7.2. Audit message format
+Audit entries are generated via specially crafted  syslog messages of the format:
+`@cee:{"TNAUDIT": {<EVENT>}}`
+
+Each audit event object must contain the following keys (as defined in NEP-041):
+
+
+.. code-block:: text
+
+    "aid" - audit identifier (GUID)
+    "vers" - JSON object with major / minor versions
+    "addr" - IP address of client (may be NULL)
+    "sess" - GUID indicating session ID (may be NULL)
+    "time" - ISO8601 timestamp of event (must not be NULL)
+    "svc" - uppercase name of service being audited
+    "svc_data" - see section 2.3 above (this should be JSON dumped)
+    "event" - uppercase name of the event
+    "event_data" - see section 2.4 above (this should be JSON dumped)
+    "success" - boolean
+
+7.3. Sample message:
+
+.. code-block:: text
+
+    @cee:{"TNAUDIT": {"aid": "5a1ad16d-7124-488d-8ec5-50e00fabc852", "vers": {"major": 0, "minor": 1}, "addr": "127.0.0.1", "user": "smbuser", "sess": null, "time": "2023-10-30 21:10:50.581658Z", "svc": "SMB", "svc_data": "{\"vers\": {\"major\": 0, \"minor\": 1}, \"service\": null, \"session_id\": null, \"tcon_id\": null}", "event": "AUTHENTICATION", "event_data": "{\"logonId\": \"0\", \"logonType\": 3, \"localAddress\": \"ipv4:127.0.0.1:445\", \"remoteAddress\": \"ipv4:127.0.0.1:48394\", \"serviceDescription\": \"SMB2\", \"authDescription\": null, \"clientDomain\": \"WORKGROUP\", \"clientAccount\": \"smbuser\", \"workstation\": \"TRUENAS\", \"becameAccount\": \"smbuser\", \"becameDomain\": \"TRUENAS\", \"becameSid\": \"S-1-5-21-2070160678-1985503795-992167086-20065\", \"mappedAccount\": \"smbuser\", \"mappedDomain\": \"WORKGROUP\", \"netlogonComputer\": null, \"netlogonTrustAccount\": null, \"netlogonNegotiateFlags\": \"0x00000000\", \"netlogonSecureChannelType\": 0, \"netlogonTrustAccountSid\": null, \"passwordType\": \"NTLMv2\", \"clientPolicyAccessCheck\": null, \"serverPolicyAccessCheck\": null, \"vers\": {\"major\": 0, \"minor\": 1}, \"result\": {\"type\": \"NTSTATUS\", \"value_raw\": 0, \"value_parsed\": \"SUCCESS\"}}", "success": true}}
```

Comment:
Might be a good idea to have this pretty-printed somehow in the docs so the example message contents can also be easily examined